### PR TITLE
Add BN_bn2binpad API and enable OpenVPN master CI testing

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

File: src/ssl_bn.c [hunk 1]
```diff
@@ -468,6 +468,56 @@ int wolfSSL_BN_bn2bin(const WOLFSSL_BIGNUM* bn, unsigned char* r)
...
+{
+    int numBytes;
+
+    WOLFSSL_ENTER("wolfSSL_BN_bn2binpad");
+
+    /* Validate parameters. */
+    if (BN_IS_NULL(bn) || (r == NULL) || (tolen < 0)) {
+        WOLFSSL_MSG("NULL bn, r, or invalid tolen error");
+        return WOLFSSL_FATAL_ERROR;
+    }
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply src/ssl_bn.c:493]: openssl doesn't guard against null deref. we do.

**Comment by dgarske** [comment]
@julek-wolfssl please address https://github.com/wolfSSL/wolfssl/pull/10148#pullrequestreview-4076599472

File: src/ssl_bn.c [hunk 2]
```diff
@@ -468,6 +468,56 @@ int wolfSSL_BN_bn2bin(const WOLFSSL_BIGNUM* bn, unsigned char* r)
...
+
+    /* Validate parameters. */
+    if (BN_IS_NULL(bn) || (r == NULL) || (toLen < 0)) {
+        WOLFSSL_MSG("NULL bn, r, or invalid toLen error");
+        return WOLFSSL_FATAL_ERROR;
+    }
+
+    /* Get the number of bytes needed to encode the big number. */
+    numBytes = mp_unsigned_bin_size((mp_int*)bn->internal);
+    if (numBytes < 0) {
```
Comment (dgarske) [inline src/ssl_bn.c:497]: **🔵 [Low] Dead code: `numBytes < 0` check can never trigger**
*🔧 NIT*  `convention`

`mp_unsigned_bin_size()` computes `(count_bits + 7) / 8` across all three math backends (sp_int.c, tfm.c, integer.c). The result is always >= 0 — it can never return a negative value. The `numBytes < 0` guard is dead code. While this is harmless defensive programming, the sister function `wolfSSL_BN_bn2bin` at line 458 does not perform this check, creating a minor inconsistency.

**Suggestion:**
```suggestion
numBytes = mp_unsigned_bin_size((mp_int*)bn->internal);
```

File: tests/api/test_ossl_bn.c [hunk 3]
```diff
@@ -323,6 +323,39 @@ int test_wolfSSL_BN_enc_dec(void)
     ExpectNotNull(BN_bin2bn(binNum, sizeof(binNum), b));
     ExpectIntEQ(BN_cmp(a, b), -1);
 
+    /* BN_bn2binpad tests */
```
Comment (dgarske) [inline tests/api/test_ossl_bn.c:326]: **🟡 [Medium] Missing test for `toLen == 0` edge case**
*💡 SUGGEST*  `test`

The tests cover invalid parameters, toLen-too-small, normal padding, exact size, and zero-value padding. However, the `toLen == 0` boundary is not tested. In the implementation, `toLen == 0` with a zero-valued BN returns 0 (success), and `toLen == 0` with a non-zero BN returns -1 (error). Both behaviors match OpenSSL's `BN_bn2binpad` and should be verified.

**Suggestion:**
```suggestion
/* toLen == 0 with zero-valued BN is valid */
        ExpectIntEQ(BN_set_word(a, 0), 1);
        ExpectIntEQ(BN_bn2binpad(a, padOut, 0), 0);
        /* toLen == 0 with non-zero BN is an error */
        ExpectIntEQ(BN_set_word(a, 2), 1);
        ExpectIntEQ(BN_bn2binpad(a, padOut, 0), -1);
```

**Review by dgarske — COMMENTED** [review]
## 🐺 Skoll Code Review

**Overall recommendation:** `APPROVE`
**Findings:** 2 total — 2 posted, 0 skipped

### Posted findings
- [Low] **Dead code: `numBytes < 0` check can never trigger** — `src/ssl_bn.c:497-499`
- [Medium] **Missing test for `toLen == 0` edge case** — `tests/api/test_ossl_bn.c:326-357`

*Review generated by Skoll via openclaw*

File: src/ssl_bn.c (hunk 2 above)
Comment (julek-wolfssl) [reply src/ssl_bn.c:497]: I think this check has to stay as long as the return type is `int`.

File: tests/api/test_ossl_bn.c (hunk 3 above)
Comment (julek-wolfssl) [reply tests/api/test_ossl_bn.c:326]: Adding.

**Comment by julek-wolfssl** [comment]
retest this please unrelated.

**Comment by julek-wolfssl** [comment]
Failures not related

**Review by JacobBarthelmeh — APPROVED** [review]

**Comment by JacobBarthelmeh** [comment]
Retest this please Jenkins
